pkgs/applications/networking/wg-bond/default.nix

```
@@ -0,0 +1,28 @@
+{ pkgs, lib, rustPlatform, fetchFromGitLab, wireguard-tools, makeWrapper }:
+rustPlatform.buildRustPackage rec {
+  pname = "wg-bond";
+  version = "0.2.0";
+  src = fetchFromGitLab {
+    owner = "cab404";
+    repo = "wg-bond";
+    rev = "v${version}";
+    hash = "sha256:04k0maxy39k7qzcsqsv1byddsmjszmnyjffrf22nzbvml83p3l0y";
+  };
+
+  cargoSha256 = "1v2az0v6l8mqryvq3898hm7bpvqdd2c4kpv6ck7932jfjyna512k";
+
+  buildInputs = [ makeWrapper ];
+  postInstall = ''
+    wrapProgram $out/bin/wg-bond --set PATH ${
+      lib.makeBinPath [ wireguard-tools ]
+    }
+  '';
+
+  meta = with lib; {
+    description = "Wireguard configuration manager";
+    homepage = "https://gitlab.com/cab404/wg-bond";
+    license = licenses.gpl3;
```

```suggestion
    license = licenses.gpl3Plus;
```
See @jtojnar 's announcement in https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206

Looks like this one can do later versions too:
https://gitlab.com/cab404/wg-bond/-/blob/master/LICENSE#L640

---

I would probably prefer to make it GPLv3 only, and upgrade manually in the future. At least for this release.

---

Yeah, that line is only instructions in the license terms, not an actual comment extending the permissions so `gpl3Only` is correct here. The project would either need to add comments to every source file, or at least clarify the license in `Cargo.toml` or `README` to allow future versions of the license.

GNU strongly recommends allowing the use of later licenses, see https://www.gnu.org/licenses/gpl-faq.html.en#VersionThreeOrLater. I would also recommend switching to that as soon as possible, since you need agreement of every code contributor to relicense code.

---

>  I would also recommend switching to that as soon as possible,

I've added that to this milestone — thanks for the tip)